### PR TITLE
Added await-signals sample

### PR DESCRIPTION
**Pull request by mfateev**
## What was changed
Added a sample demonstrating how to process signals that come out of order using `Await` and `AwaitWithTimeout`. 

## Why?
The presented pattern greatly simplifies workflow implementation in many scenarios.

### CODE REVIEW

File: await-signals/await_signals_workflow.go [hunk 1]
```diff
@@ -0,0 +1,164 @@
+package await_signals
+
+import (
+	"go.temporal.io/sdk/temporal"
+	"go.temporal.io/server/common"
```
Comment (cretz) [inline await-signals/await_signals_workflow.go:5]: We shouldn't have a dependency on the server repo in this repo at all IMO. We only did for samples that require access to the CLI plugin interface (happened before CLI moved out of server) and in a couple of cases it was used for its utilities (unnecessarily IMO).

I opened https://github.com/temporalio/samples-go/issues/237 for this. Don't have to fix now.

File: await-signals/await_signals_workflow.go [hunk 2]
```diff
@@ -0,0 +1,164 @@
...
+	}
+	log.Info("Signal1 Processed")
+
+	// Wait for Signal2
+	timeout, err := a.GetNextTimeout(ctx)
+	// No time left. At this point this cannot really happen.
+	if err != nil {
+		return
+	}
+	ok, err := workflow.AwaitWithTimeout(ctx, timeout, func() bool {
```
Comment (cretz) [inline await-signals/await_signals_workflow.go:132]: Also a good use case for your new `ReceiveWithTimeout`, but I understand that doesn't demonsrtate what you want here

File: await-signals/await_signals_workflow.go [hunk 3]
```diff
@@ -0,0 +1,164 @@
...
+	return common.MinDuration(SignalToSignalTimeout, totalLeft), nil
+}
+
+// AwaitSignalsWorkflow workflow definition
+func AwaitSignalsWorkflow(ctx workflow.Context) (err error) {
+	log := workflow.GetLogger(ctx)
+	var a AwaitSignals
+	workflow.Go(ctx, func(ctx workflow.Context) {
+		a.Listen(ctx)
+	})
```
Comment (cretz) [inline await-signals/await_signals_workflow.go:114]:
```suggestion
	workflow.Go(ctx, a.Listen)
```

**Review by cretz — APPROVED** [review]

File: await-signals/await_signals_workflow.go (hunk 1 above)
Comment (mfateev) [reply await-signals/await_signals_workflow.go]: oops, missed that it is not built-in.

**Pull request merged**